### ISSUE TITLE
Not able to delete cache by key

### ISSUE THREAD

**Issue by alisher-aituarov** [vercel/swr#2787]
# Bug report

## Description / Observed Behavior
I am making simple useSWR call passing plain text as a key. And I am trying to delete the cache for this request when component is unmounted. Here is how i am doing:
`const {cache} = useSWRConfig()

function Component(){
      const {data} = useSWR('/api/history', () => api.getHistory())
      
      useEffect(() => {}
           return () => {
                console.log('i am logged')
                cache.delete('/api/history')
           }
     ,[])
      // Rest
}

i see the log but when the component is remounted request is never made again
I also tried using mutate but did not work either

## Expected Behavior

How did you expect SWR to behave here?
Should delete the cache for a given key and make a new request. Which is not happening here
## Additional Context

SWR version - 1.3.0

**Comment by Lwdthe1** [comment]
Agreed with this. `cache.delete()` does evict the key, but the validation process is not started over from scratch when the key is used again. This causes the `data` to never be loaded again after the eviction
